gis/gccview/gcc_layer_puller.py

```diff
@@ -0,0 +1,277 @@
+from psycopg2 import connect
+from psycopg2 import sql
+from psycopg2.extras import execute_values
+import logging
+import os
+from time import sleep
+import click
+from pathlib import Path
+import configparser
+CONFIG = configparser.ConfigParser()
+from psycopg2 import connect
+
+LOGGER = logging.getLogger(__name__)
+logging.basicConfig(level=logging.INFO)
+
+# Import functions
+try:
+    from gcc_puller_function import (  get_geometry, 
+                                        to_time, 
+                                        mapserver_name, 
+                                        get_tablename, 
+                                        find_limit, 
+                                        get_fieldtype)
+except:
+    raise ImportError("Cannot import puller helper functions.")
+
+def create_table(output_table, return_json, schema_name, con):
+    """
+    Function to create a new table in postgresql for the layer (for audited tables only)
+
+    Parameter
+    ---------
+    output_table : string
+        Table name for postgresql, returned from function get_tablename
+
+    return_json : json
+        Resulted json response from calling the api, returned from function get_data
+    
+    schema_name : string
+        The schema in which the table will be inserted into
+        
+    primary_key : string
+        Primary key for this layer, returned from dictionary pk_dict
+    
+    con: Connection
+        Could be the connection to bigdata or to on-prem server
+
+    Returns
+    --------
+    insert_columm : SQL composed
+        Composed object of column name and types use for creating a new postgresql table
+    
+    excluded_column : SQL composed
+        Composed object that is similar to insert_column, but has 'EXCLUDED.' attached before each column name, used for UPSERT query
+    """
+    
+    fields = return_json['fields']
+    insert_column_list = [sql.Identifier((field['name'].lower()).replace('.', '_')) for field in fields]
+    insert_column_list.append(sql.Identifier('geom'))
+    insert_column = sql.SQL(',').join(insert_column_list)
+    
+    with con:
+        with con.cursor() as cur:
+            
+            col_list = [sql.Identifier((field['name'].lower()).replace('.', '_')) + sql.SQL(' ') + sql.SQL(get_fieldtype(field["type"])) for field in fields]
+            col_list.append(sql.Identifier('geom') + sql.SQL(' ') + sql.SQL('geometry'))
+            col_list_string = sql.SQL(',').join(col_list)
+            
+            LOGGER.info(col_list_string.as_string(con))
+            create_sql = sql.SQL("CREATE TABLE IF NOT EXISTS {schema_table} ({columns})").format(schema_table = sql.Identifier(schema_name, output_table),
+                                                                      columns = col_list_string)
+            LOGGER.info(create_sql.as_string(con))
+            cur.execute(create_sql)
+
+    return insert_column
+
+def get_data(mapserver, layer_id, max_number = None, record_max = None):
```

Comment:
Isn't this identical to gcc_layer_functions.py `get_layers` minus the centreline exception? Think you can combine them?